ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+
+# control-plane machinesets
+stderr=$(mktemp)
+stdout=$(mktemp)
+oc get controlplanemachinesets -n openshift-machine-api 1>${stdout} 2>${stderr} || true
+
+if [ -s ${stderr} ]; then
+    echo "WARN: 'controlplanemachinesets' not found, skip checking"
```

Comment:
Shall we introduce a condition to judge the cluster version for a more strict check? cc @yunjiang29 for more opinion.

Reply:
Yes, suggest adding a version check, if I remember right, `controlplanemachinesets` was introduced in 4.12, so if version < 4.12, the step can be skipped.

Reply:
@yunjiang29 I checked a 4.12 [cluster](https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Flexy-install/222057/) on **AWS**, yes, controlplanemachinesets is available. 
```
$ oc get clusterversion
NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.12.0-0.nightly-2023-07-30-150941   True        False         176m    Cluster version is 4.12.0-0.nightly-2023-07-30-150941
$ oc get controlplanemachinesets -n openshift-machine-api
NAME      DESIRED   CURRENT   READY   UPDATED   UNAVAILABLE   STATE    AGE
cluster   3         3         3       3                       Active   3h16m
$ 
```

But, for other platforms, 4.12 seems not.
Feature | Target Version | Status
-- | -- | --
[OCPSTRAT-497 Support ControlPlaneMachineSets on GCP](https://issues.redhat.com/browse/OCPSTRAT-497) | openshift-4.13 | CLOSED
[OCPSTRAT-594 Support ControlPlaneMachineSets on Azure](https://issues.redhat.com/browse/OCPSTRAT-594) | openshift-4.13 | CLOSED
[OCPSTRAT-189 Add ControlPlaneMachineSet for the Nutanix platform](https://issues.redhat.com/browse/OCPSTRAT-189) | openshift-4.14 | IN PROGRESS
[OCPSTRAT-706 Add ControlPlaneMachineSet for vSphere](https://issues.redhat.com/browse/OCPSTRAT-706) | openshift-4.15 | NEW

BTW, I followed the existing step `cucushift-installer-check-control-plane-machine-name`.

Reply:
OK, so we need to check both version and platform, I think we can refer to `CLUSTER_TYPE` https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L354